File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        mp = defaultdict(int)
        st = set()

        for num in arr:
            mp[num] += 1

        st = set(mp.values())

        x = len(mp.values())
        y = len(st)

        return x == y

File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.cs
public class Solution {
    public bool UniqueOccurrences(int[] arr) {
        var mp = new Dictionary<int, int> ();

        foreach (int num in arr) {
            if (mp.ContainsKey(num)) {
                mp[num]++;
            }
            else {
                mp[num] = 1;
            }
        }

        var st = new HashSet<int> (mp.Values);

        return mp.Values.Count == st.Count;
    }
}